docs/geps/gep-05.rst

```diff
@@ -0,0 +1,198 @@
+code:
+
+.. code-block:: yaml
+
+    rounding:
+      höchstwert_grundr_zuschlag_m:
+        2020-01-01:
+          base: 0.0001
+          direction: nearest
+          reference: §76g SGB VI Abs. 4 Nr. 4
+
+The specification of the rounding parameters starts with the key ``rounding`` at
+the outermost level of indentation. The keys are names of functions.
+
+At the next level, the ``YYYY-MM-DD`` key(s) indicate when rounding was
+introduced and/or changed. This is done in in the same way as for other policy
+parameters as described in :ref:`gep-3`. Those ``YYYY-MM-DD`` key(s) are
+associated with a dictionary of the following elements:
+
+- The parameter ``base`` determines the base to which the variables is rounded.
+  It has to be a floating point number.
+- The parameter ``direction`` has to be one of ``up``, ``down``, or ``nearest``.
+- The reference must contain the reference to the law, which specifies the
+  rounding.
+
+In the same way as other policy parameters, the rounding parameters become part
+of the dictionary ``policy_params``.
+
+A function to be rounded must be decorated with ``add_rounding_spec``. This
+decorator indicates that the output should be potentially rounded.
+``add_rounding_spec`` takes one required argument ``func_arg``, which specifies
+the (parameters) argument to the function containing the rounding as a string.
+In the above example, the rounding specification
+``höchstwert_grundr_zuschlag_m`` will be found in its argument
+``ges_rentenv_params``, Hence the ``func_arg`` argument to ``add_rounding_spec``
+has to be ``"ges_rentenv_params"``:
+
+.. code-block:: python
+
+    @add_rounding_spec(func_arg="ges_rentenv_params")
+    def höchstwert_grundr_zuschlag_m(
+        grundrentenzeiten: IntSeries, ges_rentenv_params: dict
+    ) -> FloatSeries:
+        ...
+        return out
+
+The decorator adds the attribute ``__rounding_arg__`` to the function. When
+calling ``compute_taxes_and_transfers`` with ``rounding=True``, GETTSIM will
+look for a key ``rounding`` in ``func_arg`` (here: ``ges_rentenv_params``) and
+within that, for another key containing the decorated function's name (here:
+``höchstwert_grundr_zuschlag_m``). That is, by the machinery outlined in
+:ref:`gep-3`, the following indexing of the ``policy_params`` dictionary
+
+.. code-block:: python
+
+    policy_params["ges_rentenv"]["rounding"]["höchstwert_grundr_zuschlag_m"]
+
+needs to be possible and yield the ``"base"`` and ``"direction"`` keys as
+described above.
+
+Note that GETTSIM only allows for optional rounding of functions' results. In
+case one is tempted to write a function requiring an intermediate variable to be
+rounded, the function should be split up so that another function returns the
+quantity to be rounded.
+
+Error handling
+~~~~~~~~~~~~~~
+
+In case a function has a ``__rounding_arg__``, but the respective
+parameters are missing in ``policy_params``, an error is raised.
```

Comment:
I just realized that this would also happen if rounding of a variable would change over time such that the variable isn't rounded at some point in time. For this time period, rounding parameters would be missing.

I would hence suggest that we also just test that all decorated functions have rounding parameters specified (for at least one point in time), but don't raise an error.

Reply:
This might be a case where I would it is better to require explicit setting of rounding to `None` or the like in the parameters file for periods where it does not exist. If I see the `rounding_spec` decorator, I expect it to take effect and do not want to hide required action at a distance.

That is, for the GETTSIM codebase, it is no big deal either way, I am thinking more about user-provided changes here where I would want to see errors.

Reply:
We could allow for ``direction = None``

Reply:
direction / base or both should be `None`, I'd say.

Reply:
I added a short paragraph about that.